nix/haskell.nix

```diff
@@ -4,193 +4,259 @@
 { haskell-nix
 , # Version info (git revision)
   gitrev
-, # Pre-computed package list (generated by nix/regenerate.sh) to avoid double evaluation:
-  projectPackagesExes
 , inputMap
 }:
 let
 
   inherit (haskell-nix) haskellLib;
 
-  projectPackageNames = builtins.attrNames projectPackagesExes;
-
   # This creates the Haskell package set.
   # https://input-output-hk.github.io/haskell.nix/user-guide/projects/
+  project = haskell-nix.cabalProject' ({ pkgs
+                            , lib
+                            , config
+                            , buildProject
+                            , ...
+                            }: {
+    inherit inputMap;
+    name = "cardano-node";
+    # We clean-up src to avoid rebuild for unrelated changes:
+    src = haskellLib.cleanSourceWith {
+      src = ../.;
+      name = "cardano-node-src";
+      filter = path: type: let
+        relPath = lib.removePrefix "${../.}/" path;
+      in
+        # excludes top-level directories not part of cabal project (or used by tests):
```

Comment:
Is this actually better than the builtin haskell.nix filtering? I thought it filtered things pretty well, so it only pulls in files that are literally mentioned in the cabal file?

Reply:
as discussed, this added filtering is mostly for tests that depends on the whole tree (TODO added to fix this).